### PR TITLE
Fix a regression in parsing ignore.resources for traces

### PR DESCRIPTION
**Pull request by bmermet**
### What does this PR do?

This PR addresses a regression in the parsing of the ini configuration for the trace agent introduced in https://github.com/DataDog/datadog-agent/pull/2698. 

It used to accept strings normal strings, but now requires them to be escaped with quotes. Meaning that:

```
[trace.ignore]
resource: config.get_config
```
used to work but is now broken. And given the ini parser strips quotes, the only way I've found to parse a single resource is to do:

```
[trace.ignore]
resource: '"/health"'
```

It currently works for multiple resources with the format:
```
[trace.ignore]
resource: "/health","/500"
```

But this is not backward compatible with the previous version that was:

```
[trace.ignore]
resource: "/health,/500"
```

### Motivation

This PR reverts the original behavior to avoid having a non backward compatible change.

### CODE REVIEW

File: pkg/config/legacy/converter_test.go [hunk 1]
```diff
@@ -205,6 +205,29 @@ func TestDefaultValues(t *testing.T) {
...
+	require := require.New(t)
+
+	cases := []struct {
+		config   string
+		expected []string
+	}{
+		{"r1", []string{"r1"}},
+		{`"r1","r2,"`, []string{"r1", "r2,"}},
+		{`"r1"`, []string{"r1"}},
+		{"r1,r2", []string{"r1", "r2"}},
```
Comment (arbll) [inline pkg/config/legacy/converter_test.go]: You should probably use them for every config to stay consistent and make reading easier

File: pkg/config/legacy/converter.go [hunk 2]
```diff
@@ -437,3 +435,18 @@ func buildHistogramPercentiles(agentConfig Config) []string {
...
 	return histogramPercentile
 }
+
+func splitString(s string, sep rune) ([]string, error) {
+	r := csv.NewReader(strings.NewReader(s))
+	r.TrimLeadingSpace = true
+	r.LazyQuotes = true
+	r.Comma = sep
+
+	record, err := r.Read()
```
Comment (arbll) [inline pkg/config/legacy/converter.go]: could you just return r.Read() ?

**Review by arbll — DISMISSED** [review]
A few nits, apart from them, looks good to me

File: pkg/config/legacy/converter.go (hunk 2 above)
Comment (bmermet) [reply pkg/config/legacy/converter.go]: Indeed, silly me. I copy pasted this from the `trace-agent` without giving it a thought.

**Review by arbll — DISMISSED** [review]
LGTM - For anyone looking at this PR and seeing the `critical` label: there is no impact on the version 6 of the agent. This code is shared with the datadog-trace-agent and impacts the version 5 of the agent.

**Comment by olivielpeau** [comment]
Note: this PR has a small impact on Agent 6, since it affects the `datadog-agent import` command. How does the bug affect users that that have an Agent 5 `ini` config with `resource` set under `[trace.ignore]` and run the `datadog-agent import` command?

**Comment by bmermet** [comment]
This will indeed break the `import` command for people who use it on a `ini`config affected by the bug. It will exit with this error: https://github.com/DataDog/datadog-agent/blob/master/cmd/agent/common/import.go#L67.

**Comment by arbll** [comment]
Let's add a [bugfix release note](https://github.com/DataDog/datadog-agent/blob/1e3a942091ca41b1f9ec9ff48b1b1692c295b519/docs/dev/contributing.md#reno) then.

**Review by arbll — APPROVED** [review]

**Pull request merged**